sparksql-scalapb/src/main/scala/scalapb/spark/FromCatalystHelpers.scala

```diff
@@ -42,7 +26,17 @@ trait FromCatalystHelpers {
         cmp.scalaDescriptor.fields.map { fd =>
           fieldFromCatalyst(cmp, fd, input)
         }
-      else {
+      else if (schemaOptions.sparkTimestamps && cmp.scalaDescriptor.fullName == "google.protobuf.Timestamp") {
```

Comment:
I'd prefer a more generic solution than testing for specific field types. My preference would be to have a map in schema options between `GeneratedMessageCompanion[_]` to functions that can convert that Spark datatype to protobuf and backwards. The code here can look into that map and call `Invoke` on these functions.